workspaces/adventure-pack/goodies/typescript/Iterator.prototype.tally/index.ts

```diff
@@ -0,0 +1,18 @@
+import "../Iterator.prototype.toIterable";
+import { iteratorPrototype } from "../Iterator.prototype";
+
+declare global {
+  interface Iterator<T> {
+    tally(this: Iterator<T>): Map<T, number>;
+  }
+}
+
+iteratorPrototype.tally ??= function <T>(this: Iterator<T>): Map<T, number> {
```

Comment:
We most likely don't need `??` here since we are not competing with any built-in `tally` function.